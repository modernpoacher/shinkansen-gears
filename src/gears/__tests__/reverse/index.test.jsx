import React from 'react'
import PropTypes from 'prop-types'
import renderer from 'react-test-renderer'

import Reverse from '#gears/gears/reverse'

/**
 * @param {{ to: string | { pathname: string }, children: React.ReactNode | React.ReactNode[] }} param0
 * @returns {React.JSX.Element}
 */
function MockLink ({ to, children }) {
  if (typeof to === 'string') {
    return (
      <a href={to} className='mock-link'>
        {children}
      </a>
    )
  }

  const {
    pathname
  } = to

  return (
    <a href={pathname} className='mock-link'>
      {children}
    </a>
  )
}

MockLink.propTypes = {
  to: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.shape({})
  ]),
  children: PropTypes.oneOfType([
    PropTypes.node,
    PropTypes.arrayOf(
      PropTypes.node
    )
  ])
}

jest.mock('react-router', () => {
  return {
    __esModule: true,
    Link: MockLink
  }
})

describe('#gears/gears/reverse', () => {
  it('renders', () => {
    const component = (
      <Reverse
        pathname='MOCK PATHNAME'
      />
    )

    return expect(renderer.create(component).toJSON())
      .toMatchSnapshot()
  })
})
