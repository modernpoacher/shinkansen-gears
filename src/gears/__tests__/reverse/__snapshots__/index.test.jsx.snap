// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#gears/gears/reverse renders 1`] = `
<li
  className="shinkansen-gears-reverse"
>
  <a
    className="mock-link"
    href={
      {
        "pathname": "MOCK PATHNAME",
      }
    }
  >
    Reverse
  </a>
</li>
`;
